core/embed/rust/src/ui/model_tt/component/keyboard/pin.rs

```diff
@@ -145,7 +145,7 @@ impl Component for PinKeyboard {
 
     fn event(&mut self, ctx: &mut EventCtx, event: Event) -> Option<Self::Msg> {
         if let Some(Clicked) = self.confirm_btn.event(ctx, event) {
-            return Some(PinKeyboardMsg::Confirmed);
+            return Some(PinKeyboardMsg::Confirmed(self.digits.clone()));
```

Comment:
If we keep this, is there a way to zero out the pin being cloned in a way that doesn't get optimized out?

Reply:
See above, we can get rid of PIN being stored in the message altogether. Re. reliable data zeroing, there are some crates, let's investigate how they work in a nostd env.

Reply:
https://docs.rs/zeroize/latest/zeroize/ is such a crate.

Reply:
Not storing the pin in the message anymore so zeroing isn't needed (yet). But good to know such crate exists.

Reply:
maybe you should zero out the source buffer for the PIN though?

Reply:
👍 
added the TODO

honestly, couldn't we just consume the layout object instead? `self` instead of `&mut self`
(probably not because something something reference in the Python-held object, but still something that seems semantically more correct)

Reply:
#2552